### ISSUE TITLE
[Bug] Stale cache gets never purged

### ISSUE THREAD

**Issue by Toasbi** [llfbandit/dio_cache_interceptor#58]
Hey 
so I was experimenting with the CacheOptions and forceCache and it turns out that a recent change broke the purging of stale entries.
https://github.com/llfbandit/dio_cache_interceptor/commit/cca456e6e8e57cb751955014f7242c0435c81c76?diff=split#diff-64d31cdb5c921267241ca3b129a926e65686a33f36c8e393776643f70dd38867L169

It seems like the problem is that once you set the global CacheOptions to have a maxStale value, the if condition mentioned above is always false and stale entries do not get purged. This still works with version 3.1.0 though.

I think the easiest way to reproduce this is add this 2 lines here:
https://github.com/llfbandit/dio_cache_interceptor/commit/cca456e6e8e57cb751955014f7242c0435c81c76?diff=split#diff-25cfd2bce1cd1995fee8c59ec60b98c2ba1936d3e00b046e841238c543108b83R66

```
    fromNetwork = resp.extra[CacheResponse.fromNetwork];
    expect(fromNetwork, isTrue);
```

This should ensure that the response is not coming from the expired cache.

Hope this helps

**Comment by llfbandit** [comment]
Thanks for the report!
Looking into it now.

**Comment by llfbandit** [comment]
This is now fixed on master branch.
If you set `maxStale` on default options, you still must provide dedicated options on the request to avoid postponing.

I changed the signature of `CacheOptions.copyWith` to ease those manipulations:
```dart
options.copyWith(maxStale: Nullable(null));
```
Feedback welcomed.

**Comment by Toasbi** [comment]
Thanks for the fast reaction!
Tested it with the latest branch and seems to work now.

**Comment by llfbandit** [comment]
Released in v3.2.3

**Comment by beerline** [comment]
@llfbandit  
Thank u for such a useful library 

I faced with same issue on version `3.5.0`
if `maxStale` not set cache will never purged

Example:
Server sent: 'Cache-Control: max-age=60'
The resource is considered fresh for 60 seconds (1 minutes).

Expected behavior:
1. during 60 seconds after response return cache data
2. after 60 seconds consider cache as staled and return connection error

Reality:
cache never purged.
